lib/index.js

```diff
@@ -55,6 +56,55 @@ exports.plugin = {
             server.ext('onPreStart', internals.initialize);
             server.ext('onPostStop', internals.stop);
 
+            // Expose an hpal command for schwifty-migration
+            server.expose('commands', {
+                'migrate:diff': (initializedServer, args, rootDir, ctx) => {
+
+                    const SchwiftyMigrateDiff = require('./schwifty-migrate-diff').get();
+
+                    if (!SchwiftyMigrateDiff) {
+                        throw new ctx.DisplayError('To use this command please first install `npm install --save-dev schwifty-migrate-diff`');
+                    }
+
+                    const [mode, migrationName, argsMigrationsDir] = args;
+
+                    const realmsWithMigrationsDir = internals.state(initializedServer.realm).collector.realmsWithMigrationsDir;
+
+                    const absRootDir = Path.isAbsolute(rootDir) ?
+                        Path.normalize(rootDir) :
+                        Path.join(process.cwd(), rootDir);
+
+                    const realmsWithMigrationsInRoot = Array.from(realmsWithMigrationsDir).reduce((collector, realm) => {
+
+                        const migrationsDir = internals.state(realm).migrationsDir;
+                        if (migrationsDir.includes(absRootDir)) {
+                            collector.push(realm);
+                        }
+
+                        return collector;
+                    }, []);
+
+                    Hoek.assert(realmsWithMigrationsInRoot.length > 0, `At least 1 migrations directory must be present inside ${absRootDir}`);
+                    Hoek.assert(realmsWithMigrationsInRoot.length === 1, `Only 1 migrations directory supported per project at this time. Please remove one of [${realmsWithMigrationsInRoot.map((realm) => realm.plugins.schwifty.migrationsDir)}]`);
+
+                    const [migrationRealm] = realmsWithMigrationsInRoot;
+
+                    const serverModels = initializedServer.models();
+                    const allModels = Object.keys(serverModels).map((modelName) => serverModels[modelName]);
+                    const realmKnex = internals.state(migrationRealm).knex;
+                    const models = Array.from(allModels).filter((model) => model.knex() === realmKnex);
+                    const modelsArr = Object.keys(models).map((key) => models[key]);
+
+                    return Util.promisify(SchwiftyMigrateDiff.genMigrationFile)({
+                        models: modelsArr,
+                        migrationsDir: argsMigrationsDir || internals.state(migrationRealm).migrationsDir,
```

Comment:
You have `migrationsDir` from above, don't think you need to pull it off state again.

Reply:
That one is in a loop